contracts/src/WarpProtocolRegistry.sol

```diff
@@ -13,6 +13,7 @@ import "@openzeppelin/contracts/utils/Address.sol";
  */
 struct ProtocolRegistryEntry {
     uint256 version;
+    address destinationAddress;
```

Comment:
It seems a little odd to add `destinationAddress` into the `ProtocolRegistryEntry` struct, since the entry itself is still just a `version->protocolAddress` map. 

It could be a bit easier to understand if had a `UpdateRegistryMessage` struct that had a `ProtocolRegistryEntry` member, as well as the destination address of the message. This way the constructor wouldn't need a "destinationAddress" value either, since it's not used.

cc @minghinmatthewlam in case you have thoughts here.

Reply:
yup I was thinking the same so sgtm. Only nit is I prefer more similar naming, we have `addProtocolVersion` function and `AddProtocolVersion` event, so would consider something similar like `AddProtocolVersionInput`.

Reply:
actually now that I think about it we don't need the second struct, we can decode two arguments one of them being `destinationAddress` instead 

```
        (ProtocolRegistryEntry memory entry, address destinationAddress) = abi.decode(
            message.payload,
            (ProtocolRegistryEntry, address)
        );
```

Reply:
fixed